contracts/ServiceNodeRewards.sol

```diff
@@ -755,11 +797,40 @@ contract ServiceNodeRewards is Initializable, Ownable2StepUpgradeable, PausableU
     /// the owner.
     /// @param newMax The new maximum non-signer threshold
     function setBLSNonSignerThresholdMax(uint256 newMax) public onlyOwner {
-        require(newMax > 0, "The new BLS non-signer threshold must be non-zero");
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
         blsNonSignerThresholdMax = newMax;
         emit BLSNonSignerThresholdMaxUpdated(newMax);
     }
 
+    /// @notice Max Claim amount to use in the Exponential Moving Average check
+    /// before allowing the user to claim. If the claimed amount over 24 hours
+    /// exceeds this then the claim will fail
+    /// @param newMax The new maximum non-signer threshold
+    function setClaimEMAThreshold(uint256 newMax) public onlyOwner {
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
+        claimEMAThreshold = newMax;
+        emit ClaimEMAThresholdUpdated(newMax);
+    }
+
+    /// @notice Updates the Exponential Moving Average (EMA) with a new value
+    /// @dev This function calculates a time-weighted EMA using the provided new value and current time
+    /// @param ema The TimeWeightedEMA struct to update
+    /// @param newValue The new value to incorporate into the EMA.
+    /// @param currentTime The current timestamp, usually `block.timestamp`
+    /// @custom:formula EMA = EMA_old + α * (newValue - EMA_old)
+    /// where α = 1 - 0.5 ^ (timeDiff / halfLife)
+    function updateEMA(TimeWeightedEMA storage ema, uint256 newValue, uint256 currentTime) internal {
+        uint256 timeDiff = currentTime - ema.lastUpdateTime;
+        uint256 SCALING_FACTOR = 262144; // 2^18
+        if (timeDiff > 0) {
```

Comment:
This condition introduces a vulnerability: I could submit multiple withdrawals in the same transaction (or same block) and only the first would update the EMA, so I could construct another contract that calls this twice in the same transaction: first with an acceptable value, and then again with a large value, to exceed the limit.